src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -55,8 +58,13 @@ private sealed class SequenceEqualComparer : IEqualityComparer<List<Section>>
         {
             public static SequenceEqualComparer Instance { get; } = new SequenceEqualComparer();
 
-            public bool Equals(List<Section> x, List<Section> y)
+            public bool Equals([AllowNull] List<Section> x, [AllowNull] List<Section> y)
             {
+                if (x is null || y is null)
+                {
+                    return x is null && y is null;
+                }
```

Comment:
Can simplify to 

```
if (x is null)
{
  return y is null;
}
```

Reply:
I don't think that's tight enough. If only `y` is null, then we would NRE  on the dereference on next line.

Reply:
Doh that's right.